sql/src/test/java/io/crate/memory/OnHeapMemoryManagerTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.memory;
+
+import io.netty.buffer.ByteBuf;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
+
+public class OnHeapMemoryManagerTest {
+
+    @Test
+    public void test_allocated_bytebuffers_are_accounted() {
+        var bytes = new AtomicLong();
+        var memoryManager = new OnHeapMemoryManager(bytes::addAndGet);
+
+        memoryManager.allocate(20);
+        assertThat(bytes.get(), is(20L));
+        memoryManager.close();
+        assertThat(bytes.get(), is(20L));
```

Comment:
maybe add a comment that this is expected behaviour as the highlevel accounting longconsumer (circuitbreaker) must be released to release the accounted bytes instead? 
Or maybe better to add a class description to the `OnHeapMemoryManager`?